.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,55 @@
+name: deploy
+
+on:
+  pull_request:
+  push:
+    branches: [main, '*-assessment-year']
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  publish-docker-image:
```

Comment:
This step is called `publish-docker-image` rather than `deploy` because I'm anticipating that we will add further jobs that depend on this one to run the model using the Docker image in https://github.com/ccao-data/model-res-avm/issues/23.